libcontainer/utils/utils_unix.go

```diff
@@ -67,3 +68,16 @@ func NewSockPair(name string) (parent *os.File, child *os.File, err error) {
 	}
 	return os.NewFile(uintptr(fds[1]), name+"-p"), os.NewFile(uintptr(fds[0]), name+"-c"), nil
 }
+
+// RunWithUmask runs a function f with umask mask.
+// It does so while locking the OS thread, as umask
+// operates on threads, and this avoid potentially leaking umask(0000)
+// to other threads.
+func RunWithUmask(mask int, f func() error) error {
+	runtime.LockOSThread()
+	oldMask := unix.Umask(mask)
+	err := f()
```

Comment:
If we're being very pedantic, this won't handle panics -- you probably want to do the unlock and resetting of the umask with `defer`s (which are much faster in modern versions of Go than they were in the earlier days).

Reply:
pedantry works for me!